pkg/connector/connector.go

```diff
@@ -102,45 +114,73 @@ func (c *Okta) Metadata(ctx context.Context) (*v2.ConnectorMetadata, error) {
 }
 
 func (c *Okta) Validate(ctx context.Context) (annotations.Annotations, error) {
-	l := ctxzap.Extract(ctx)
-	token := newPaginationToken(defaultLimit, "")
-
-	_, respCtx, err := getOrgSettings(ctx, c.client, token)
-	if err != nil {
-		return nil, fmt.Errorf("okta-connector: verify failed to fetch org: %w", err)
+	if c.apiToken != "" {
+		l := ctxzap.Extract(ctx)
+		token := newPaginationToken(defaultLimit, "")
+
+		_, respCtx, err := getOrgSettings(ctx, c.client, token)
+		if err != nil {
+			return nil, fmt.Errorf("okta-connector: verify failed to fetch org: %w", err)
+		}
+
+		_, _, err = parseResp(respCtx.OktaResponse)
+		if err != nil {
+			return nil, fmt.Errorf("okta-connector: verify failed to parse response: %w", err)
+		}
+
+		if respCtx.OktaResponse.StatusCode != http.StatusOK {
+			err := fmt.Errorf("okta-connector: verify returned non-200: '%d'", respCtx.OktaResponse.StatusCode)
+			l.Error("Invalid Status Code from Verity", zap.Error(err))
```

Comment:
I don't think the log message is needed here. The error is getting returned so it will get logged.